.github/workflows/macOS-tests.yml

```diff
@@ -7,6 +7,8 @@ on:
       - develop
       - hotfix
       - unstable
+      # Temporary develop-X.Y.Z branches may be added and removed from here as we release new versions
+      - develop-4.0
```

Comment:
This given section was added in the past, where I'd simply pushes some changes directly at the branch without any review. While it could be useful to push some code directly, I assume that it is so only in case of urgency, like the prequel merge corrupt branch somehow and now it's needed to be fixed asap. 

Long story short: I suggest to not add any new branches here, and consider to do something with this whole section in future, simultaneously hardening push rules to `develop` and `master`.

Reply:
I'd remove this section completely. `develop` and `master` branches should be protected and direct pushes to these must be prohibited for everyone.

What would you say if I remove this block with `push:` rule? All CI **Swift test runs** must be complete before we merge anything.

Reply:
This is hard.
Actually as for now there's a several tasks that i'm doing locally and them pushing it to the `master`. One is merge `devel` to `master`, but this one i certainly can do through PR merge option by GitHub. The second one is updating `podspec` within `master` for every given release. Apparently i can do the same way as i'm doing it now, coz i having admin rights, but i don't like that. The perfect state would be if it was handled by GitHub actions as well, i mean creating release, or at least merging it. But for updating `podspec` i have doubts that this is possible, because of Cocoapods modularity restrictions: e.g. I have to push into pods Web3Core at first and then i could push there Web3Swift itself only after a day or so. So if you have any ideas how we can handle this edge case I'd be happy to hear.

Anyway my state is that cocoapods is an unloved child here, so if we have to impair it in anyway in sake of something valuable we should do that.

Reply:
Okay, then to make it easy I'll remove `unstable` and whatever else I've added to the `push` rule and we can discuss CocoaPod updates in a separate issue.